### PR TITLE
doc: dts: Fix outdated identifier list for /chosen and improve related docs

### PR DESCRIPTION
**Pull request by ulfalizer**
Updating the list of identifiers generated for `/chosen` was overlooked after commit 8ce0cf0126 ("kconfig: Convert device tree chosen properties to new kconfigfunctions"). Update the list, and also add some missing documentation.

Commits:

```
doc: dts: Fix outdated list of generated macro names

The *_ON_DEV_NAME macros generated from /chosen properties changed
prefix from DT_* to CONFIG_* in commit 8ce0cf0126 ("kconfig: Convert
device tree chosen properties to new kconfigfunctions"), but the
devicetree documentation still lists the old names. Update the
documentation with the correct anmes.

Also remove an outdated reference to
DT_SRAM_SIZE/DT_SRAM_BASE_REFERENCE, and give a complete list of
generated macros.
```

```
doc: dts: Add missing documentation for some /chosen properties

Not all /chosen properties were documented. Add the missing ones along
with the macros generated for them.

Found with some grepping for '\<zephyr,.*='.
```

```
doc: dts: Explain better how devicetree and Kconfig fit together

It's cryptic that some identifiers for devicetree-related stuff start
with DT_*, while others start with CONFIG_*. Explain what's going on,
and link to the Kconfig preprocessor documentation.

Also remove an early mention of bindings that might be confusing.
Bindings are much more about checking devicetree conformance than about
controlling output, though they do some of that too.
```

```
doc: dts: Linkify Kconfig syms and explain a Kconfig reference gotcha

Turn the CONFIG_* identifiers in the /chosen table into links to the
Kconfig reference.

Also explain why devicetree information doesn't show up in the Kconfig
reference.
```

### CODE REVIEW

File: doc/guides/dts/index.rst [hunk 1]
```diff
@@ -470,33 +473,58 @@ to ``uart1`` in this board:
...
+   documentation, the Kconfig symbol reference pages linked below do not
+   include information derived from devicetree. Instead, you might see e.g. an
+   empty default:
+
+   .. code-block:: none
+
+      default "" if HAS_DTS
+
+   To see how the preprocessor is used for a symbol, look it up directly in the
+   :file:`Kconfig` file where it is defined instead. The referenc page for the
```
Comment (dbkinder) [inline doc/guides/dts/index.rst]: fix misspelling of ``reference``

**Review by dbkinder — CHANGES_REQUESTED** [review]
One misspelling to fix, otherwise LGTM.  (I notice though that CI hasn't reported the doc build results.)

File: doc/guides/dts/index.rst (hunk 1 above)
Comment (ulfalizer) [reply doc/guides/dts/index.rst]: Fixed

**Review by dbkinder — APPROVED** [review]

**Comment by ulfalizer** [comment]
@galak @pabigot 
Ping

**Review by carlescufi — APPROVED** [review]

**Pull request merged**